tt_metal/tt_stl/reflection.hpp

```diff
@@ -807,6 +807,7 @@ struct get_first_object_of_type_t<std::vector<T>> {
         for (auto& tensor : value) {
             return get_first_object_of_type<object_t>(tensor);
         }
+        throw std::out_of_range("No such element");
```

Comment:
Picked something arbitrary to satisfy the compiler's complaint that the f'n may exit without returning a value.  Looking to the owners for context re: are we guarnateed that this will never happen, and can we hint to the compiler as much?  Failing that, what is the desired outcome of this f'n?